pkg/pool-weighted/contracts/smart/IndexPool.sol

```diff
@@ -93,11 +92,44 @@ contract IndexPool is BaseWeightedPool {
         // Immutable variables cannot be initialized inside an if statement, so we must do conditional assignments
         _tokens = tokens;
 
-        for(uint i = 0; i < numTokens; i++){
+        for (uint256 i = 0; i < numTokens; i++) {
             scalingFactors.push(_computeScalingFactor(tokens[i]));
         }
     }
 
+    function reweighTokens(address[] memory tokens, uint96[] memory normalizedWeights) public {
+        uint256 numTokens = tokens.length;
+        InputHelpers.ensureInputLengthMatch(numTokens, normalizedWeights.length);
+
+        uint256 normalizedSum = 0;
+        for (uint8 i = 0; i < numTokens; i++) {
+            uint256 normalizedWeight = normalizedWeights[i];
+
+            normalizedSum = normalizedSum.add(normalizedWeight);
+        }
+        _require(normalizedSum == FixedPoint.ONE, Errors.NORMALIZED_WEIGHT_INVARIANT);
+    }
+
+    function reindexTokens(
+        address[] memory tokens,
+        uint96[] memory normalizedWeights,
+        uint256[] memory minimumBalances
+    ) external {
+        uint256 numTokens = tokens.length;
+        InputHelpers.ensureInputLengthMatch(numTokens, normalizedWeights.length);
```

Comment:
We should do this in one function, there is a version of `ensureInputLengthMatch` for 3 variables

Reply:
It is called `ensureInputLengthMatch` as well jik